contracts/aux/auto-alex-v3-wrapped.clar

```diff
@@ -0,0 +1,107 @@
+	(begin
+		(try! (is-dao-or-extension))
+		(ok (var-set token-symbol new-symbol))))
+
+(define-public (set-token-uri (new-uri (optional (string-utf8 256))))
+	(begin
+		(try! (is-dao-or-extension))
+		(ok (var-set token-uri new-uri))))
+
+;; public functions
+
+(define-public (transfer (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
+	(begin
+		(asserts! (or (is-eq tx-sender sender) (is-eq contract-caller sender)) err-unauthorised)
+		(try! (ft-transfer? auto-alex-v3-wrapped amount sender recipient))
+		(print { type: "transfer", amount: amount, sender: sender, recipient: recipient, memo: memo })
+		(ok true)))
+
+(define-public (transfer-fixed (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
+    (transfer amount sender recipient memo))
+
+(define-public (mint (amount uint) (recipient principal))
+	(begin 
+		(asserts! (or (is-eq tx-sender recipient) (is-eq contract-caller recipient)) err-unauthorised)				
+		(try! (ft-mint? auto-alex-v3-wrapped (get-tokens-to-shares amount) recipient))
+		(contract-call? .auto-alex-v3 transfer amount recipient (as-contract tx-sender) none)))
+
+(define-public (mint-fixed (amount uint) (recipient principal))
+    (mint amount recipient))
+
+(define-public (burn (amount uint) (sender principal))
+	(begin
+		(asserts! (or (is-eq tx-sender sender) (is-eq contract-caller sender)) err-unauthorised)
+		(as-contract (try! (contract-call? .auto-alex-v3 transfer (get-shares-to-tokens amount) tx-sender sender none)))
+		(ft-burn? auto-alex-v3-wrapped amount sender)))
+
+(define-public (burn-fixed (amount uint) (sender principal))
+    (burn amount sender))
+
+;; read-only functions
+
+(define-read-only (is-dao-or-extension)
+	(ok (asserts! (or (is-eq tx-sender .lisa-dao) (contract-call? .lisa-dao is-extension contract-caller)) err-unauthorised)))
+	
+(define-read-only (get-name)
+	(ok (var-get token-name)))
+
+(define-read-only (get-symbol)
+	(ok (var-get token-symbol)))
+
+(define-read-only (get-token-uri)
+	(ok (var-get token-uri)))
+
+(define-read-only (get-decimals)
+	(ok token-decimals))
+
+(define-read-only (get-balance (who principal))
+	(ok (ft-get-balance auto-alex-v3-wrapped who)))
+
+(define-read-only (get-balance-fixed (who principal))
+	(get-balance who))
+
+(define-read-only (get-total-supply)
+	(ok (ft-get-supply auto-alex-v3-wrapped)))
+
+(define-read-only (get-total-supply-fixed)
+	(get-total-supply))
+
+(define-read-only (get-share (who principal))
+	(ok (get-shares-to-tokens (unwrap-panic (get-balance who)))))
+
+(define-read-only (get-total-shares)
+	(contract-call? .auto-alex-v3 get-balance (as-contract tx-sender)))
+
+(define-read-only (get-tokens-to-shares (amount uint))
+	(if (is-eq (get-total-shares) (ok u0))
+		amount
+		(/ (* amount (unwrap-panic (get-total-supply))) (unwrap-panic (get-total-shares)))))
+
+(define-read-only (get-shares-to-tokens (shares uint))
+	(if (is-eq (get-total-supply) (ok u0))
```

Comment:
use a variable here

Reply:
I am not sure how it can be done. Please advise?

Reply:
`get-total-supply` is `(ft-get-supply auto-alex-v3-wrapped)`, so think we want to keep it as is.

Reply:
We call `total-supply` twice in this function

Reply:
https://github.com/lisalab-io/liquid-stacking/pull/56/commits/389890350ed3123517d8aeaebd3222b1669c3661